homeassistant/components/yamaha_musiccast/select.py

```diff
@@ -59,4 +59,6 @@ def options(self):
     @property
     def current_option(self):
         """Return the currently selected option."""
-        return self.capability.options[self.capability.current]
+        return self.capability.options.get(
+            self.capability.current, self.capability.current
+        )
```

Comment:
The `options` property above is based on `list(self.capability.options.values())`

This change makes it possible it returns `self.capability`, even though, thus not a valid option (as in, not in the options list?)?

Reply:
Yes, we could also return `None` as default value, but I thought it would be better to return here the actual value, even if it is not a valid option, because there is a check in the `state` property of the [SelectEntity](https://github.com/home-assistant/core/blob/e6d0a8894fff03211e104c1ea5ab19e45e732b34/homeassistant/components/select/__init__.py#L91) anyway, which returns `None` if the current option is not in the list of options

Reply:
In that case it should be just:

```suggestion
        return self.capability.options.get(self.capability.current)
```

There is no reason to return something we already know will be invalid.

Reply:
Fine for me. I just thought it would be better to keep the information as long as possible, but indeed nobody would be able to read it, so their is not really a use case for keeping it.